### PR TITLE
sample/quote: Add Makefile and README

### PR DESCRIPTION
**Pull request by binxing**

### CODE REVIEW

File: samples/quote/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ACONCLI = aconcli
+OPENSSL = openssl
+DOCKER  = docker
+GO      = go
+
+all: server.json client/sampleclient
+
+server.json: server/Dockerfile signer.pem signer.cer | .acon/
+	$(DOCKER) build -t sampleserver -f $< $(PWD)/../..
+	$(ACONCLI) generate -i sampleserver $@
```
Comment (xiangquanliu) [inline samples/quote/Makefile:13]: I remember that "writableFS" needs to be changed to "true".

File: samples/quote/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Remote Attestation Sample Code
+
+This directory contains sample source code to demonstrate how remote attestation works in ACON containers. More information on TDX remote attestation can be found at https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/TDX_Quoting_Library_API.pdf.
+
+This sample is comprised of a server and a client that communicate with each other over TCP. The server is an ACON container that generates TDX quotes upon requests from the client, which is a command line application running in the host.
+
+The client works with the server to retrieve a TD quote, along with the measurement logs and additional attestation data. It then verifies the quote, checks if RTMR values in the quote match those calculated from the measurement logs and if the server has attached the expected attestation data, and finally decodes/displays everything it receives . Addtionally, the quote is written to `quote.bin`, while RTMR logs and report data are written to `quote.json`.
```
Comment (xiangquanliu) [inline samples/quote/README.md:7]: 1. A space between 'receives' an '.'
2. 'Addtionally' -> 'Additionally'

File: samples/quote/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Simply type `make` to build both the server and the client. `docker`, `openssl`, and [`aconcli`](../../doc/GettingStarted.md#building-aconcli) are required in the build process.
+
+Running the sample requires a TDX enabled platform.
+
+- The server must be started first, by
+
+  ```sh
+  TCP_PORT=5555 ATD=1 ATD_KERNEL=/path/to/vmlinuz ATD_RD=/path/to/initrd.img aconcli run -ni -c:$TCP_PORT server.json
```
Comment (xiangquanliu) [inline samples/quote/README.md:16]: Does it need a TCP port mapping between server and client?

File: samples/quote/Makefile (hunk 1 above)
Comment (binxing) [reply samples/quote/Makefile:13]: Does quote_server need to write disk files? If no, then we don't need writable FS.

File: samples/quote/README.md (hunk 2 above)
Comment (binxing) [reply samples/quote/README.md:7]: Good catch!

File: samples/quote/README.md (hunk 3 above)
Comment (binxing) [reply samples/quote/README.md:16]: You are right! Which port?

Comment (binxing) [reply samples/quote/README.md:16]: @xxu36 and @xiangquanliu, I added `ATD_TCPFWD=8080:8085`. Let me know if I'm wrong.

Comment (xiangquanliu) [reply samples/quote/README.md]: I think that's right.

File: samples/quote/Makefile (hunk 1 above)
Comment (xiangquanliu) [reply samples/quote/Makefile]: No. Seems there is a bug of acond.

Comment (xiangquanliu) [reply samples/quote/Makefile]: @binxing, is `/run/user/0` a must directory for readonly file system?

Comment (binxing) [reply samples/quote/Makefile]: `/run` is a separate tmpfs mount that shouldn't be affected by `writableFS`.

Comment (xiangquanliu) [reply samples/quote/Makefile]: I mean `/run/user/*`, not `/run`.

File: samples/quote/README.md (hunk 3 above)
Comment (xxu36) [reply samples/quote/README.md]: yes, currently we are using this mapping

**Review by xxu36 — APPROVED** [review]

File: samples/quote/Makefile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	$(ACONCLI) sign -k signer.pem -c signer.cer $@
+
+%.pem:
+	$(OPENSSL) ecparam -genkey -name secp384r1 -out $@
+
+%.cer: %.pem
+	$(OPENSSL) req -x509 -sha384 -key $< -out $@ -outform der -subj /CN=self-signed-$<
+
+client/sampleclient:
+	$(GO) -C $(@D) build -v
```
Comment (xiangquanliu) [inline samples/quote/Makefile:24]: Not sure if it is better to build a static linked binary. For example, to build it in development environment and test in EMR environment - may cause link errors.

**Pull request merged**